src/main/java/dan200/computercraft/shared/peripheral/generic/data/ItemData.java

```diff
@@ -68,6 +75,65 @@
             }
         }
 
+
+        /*
+         * Used to hide some data from ItemStack tooltip.
+         * @see https://minecraft.gamepedia.com/Tutorials/Command_NBT_tags
+         * @see ItemStack.getTooltip(@ Nullable PlayerEntity playerIn, ITooltipFlag advanced)
+         */
+        int hideFlags = (tag != null) ? tag.getInt( "HideFlags" ) : 0;
+
+        if ( stack.isEnchanted() )
+        {
+            List<Map<String, Object>> enchants;
+            if ( (hideFlags & 1) == 0 )
+            {
+                /*
+                 * Mimic the EnchantmentHelper.getEnchantments(ItemStack stack) behavior without special case for Enchanted book.
+                 * I'll do that to have the same data than ones displayed in tooltip.
+                 * @see EnchantmentHelper.getEnchantments(ItemStack stack)
+                 */
+                enchants = enchantMapToList( EnchantmentHelper.func_226652_a_( stack.getEnchantmentTagList() ) );
```

Comment:
I'd be inclined to just use `EnchantmentHelper` and merge the books and normal enchantments. The two are displayed the same in the UI, and it'l pretty rare for books to have _actual_ enchantments

Reply:
I'll keep it since enchanted book with hideFlags=1 still display they embedded enchants. But i can merge the two outputs ;)